### DISCUSSION TITLE
TestFlight Version 1.4.4 (all Builds)

### DISCUSSION THREAD

**Discussion by hape42** [hape42/dailygammon#67] [General]
The communication via TestFlight or from each tester to me via email is way too one-sided. It would be better for the project if all testers had the opportunity to see what problems other testers are having. maybe then we can find the causes more quickly together and get to a release for all users more quickly.

When you describe problems, the device and the IOS version should always be mentioned.

Translated with www.DeepL.com/Translator (free version)

**Comment by hape42** [comment]
Device **iPhone XR** IOS **16.1.2**
Simulator **iPhone 12** IOS **15.4**

I played matches against different opponents. There were no problems while playing. 

With the display zoom turned on, the buttons in the match are now also displayed correctly.

**Comment by C4rsten** [comment]
I have some issues with switching color themes. First of all, it is very slow and afterwards some buttons are missing text and other are even missing their background. I can provide screenshots later. Using dark mode on iPad OS 15.

**Reply by hape42** [reply]
Forget to tell this version I sent by testFlight is for iPhone only.😇

The version that will also work for iPad will come in the next few days.
You can always load  the version from the Appstore. 

You can also install different versions from earlier testFlight.

**Reply by hape42** [reply]
change of theme is slow? or has the app become slower in general?

**Reply by C4rsten** [reply]
My iPad is old. Basically I think what happens is this: Since changing the theme does not react immediately I tapped it a few more times. Then the app is stuck for around 10 seconds, maybe because it sets the same theme several times.

**Reply by kagsteiner** [reply]
Having the same slow change time on my iPad Pro and iPhone, both pretty new. iPad Pro somewhere between 4 and 8 secs to change. I feel it's faster on the iPhone and particularly slow if the board is viewed while changing (which can happen only in iPad mode).

**Reply by hape42** [reply]
just tested my **iPad 7** IOS **15.6.1**

Theme change less than 1 second 🤔

**Reply by kagsteiner** [reply]
Strange, maybe this is different in iOS 16 (my iPad Pro 12,9, 3rd gen) runs iPadOS 16.2. Did you do multiple changes? I've seem to remember that the first switch sometimes was fast for me...

Here's the last few tests (all: I am in the match overview - top page, click settings, click board styles, scroll down, click once on a new board):
- 12 seconds to switch to unicorn
- 9 seconds to switch to mono
- 8 secs to classic original
- 12 to wood
- 8 back to classic original

(doesn't seem to be a larger volume of graphics data of the new boards as classic original is also slow)

**Reply by kagsteiner** [reply]
iphone (iPhone 12 pro max running iOS 16.2) (yes my employer forces me to always update ASAP :-) )

- switch to unicorn: crashed aft 7 secs (???)
- app restart, switch to mono: 1 sec
- switch back to unicorn, 1 sec (probably 1.5)
- switch to wood: 1 sec
- switch to classic original: <1 sec
- Further switches: instantaneous

This is a real WTF moment now :-)

**Reply by hape42** [reply]
very fast after crash / restart? 🤔
that could be a memory leak

**Reply by kagsteiner** [reply]
might be. Afterwards did a sequence of
top page -> settings -> board scheme -> select different scheme -> back -> back -> select a match to see the board drawn in the new scheme -> back to start

The time to switch to the new scheme increased from 1 to 3 seconds, but then seemed to stay around that time...

**Reply by kagsteiner** [reply]
Tried the same in the simulator; time also increases. Playing around in the CPU traces without much success. Much time spent in an afterCommitHandler ;-) (now that is certainly helpful)

**Reply by kagsteiner** [reply]
Seems very much time is spent in
<img width="988" alt="Bildschirm­foto 2023-01-05 um 15 16 08" src="https://user-images.githubusercontent.com/25194620/210800894-d31ddded-2a64-4755-8870-065e2a7a5795.png">

-[BoardSchemeVC tableView:didSelectRowAtIndexPath:]

**Comment by kagsteiner** [comment]
Test works fine on my iPhone 12 pro max. Like the new slightly 3dish button design.

**Comment by kagsteiner** [comment]
Small bug: If I do this:
1. be in top page
2. settings
3. board scheme
4. in board scheme, press okay

Then I'm back in the settings but the buttons for "board scheme" and "DailyGammon preferences" contain no text.

**Reply by hape42** [reply]
Thanks for the feedback.

I have found the error and fixed it.

A new version should come today via testFlight.

**Comment by b-dubbs** [comment]
I'd love to help testing -- I have an older iPhone (8) but am using the current iOS (16.2)

**Reply by hape42** [reply]
Thank you very much. For the invitation to testFlight I need your email address. You can also send it to dg@hape42.de

**Comment by MacCool2** [comment]
I want to help testing, this is my e-mail address: timinity@gmail.com

**Comment by jaggington** [comment]
I want to help testing, this is my e-mail address:
jagrenier67@gmail.com

**Reply by RickBlaine3** [reply]
I want to help testing again, this is my email address: billga3@gmail.com AKA RickBlaine

**Comment by Smiles2go** [comment]
With builds 318 and 324 I'm getting lots of Ooops Unexpected errors like: 

![image](https://user-images.githubusercontent.com/121371957/210942550-a0917a2d-3be2-4f76-bb8c-5f0bf3f594fa.png)

Going to Top Page and trying to advance with those games, same error recurs. At first thought possibly related to particular opponent names but it is not. 

Going to DG web, the stuck games all appeared to have a (Predicted Result) showing. Hitting Next from there and returning to DG app, match continues normally for those opponents. 

![image](https://user-images.githubusercontent.com/121371957/210942975-c818e9fc-099e-46b6-8ef3-e9b309968b30.png)

**Reply by hape42** [reply]
that is a really good hint.
I will try to reproduce this. If I succeed, I have a chance to see in the debugger what happens there

**Reply by hape42** [reply]
I do not succeed in reproducing that 
my preference are:
![Bildschirm­foto 2023-01-06 um 10 00 25](https://user-images.githubusercontent.com/2091581/210968144-fcca7035-296d-4860-a4b6-25ccca8be221.png)

maybe we have a difference?

**Reply by Smiles2go** [reply]
Those are my usual settings. Also: 

![image](https://user-images.githubusercontent.com/121371957/211021845-b03025de-6adc-4420-ae11-272725fa838b.jpeg)

I tried changing some to see if it might help but didn't. Changed back now so I match yours. 

No Oooops errors yet today. Next time I get some (probably later today) I will also try with iPad.

**Reply by Smiles2go** [reply]
Hit several more Ooooops Unexpected today. iPhone 14pro, iOS 16.2, DG build 324. 

Looking in DG web, none today show "Predicted". Also, none have shown any problems on iPad DG build 324. 

I was able to capture iPhone screen that appears very briefly after hitting Oooops button for Top Page, showing "-1" active matches.

![image](https://user-images.githubusercontent.com/121371957/211069104-4a98d84b-28b3-415d-a8a8-51dec652795e.png)

Getting back to Top Page shows correct # Matches where you can move.

**Comment by ogarbisu** [comment]
I want to help testing, this is my e-mail address ogarbisu@gmail.com

**Comment by kagsteiner** [comment]
Strange crash (sent mail to support) - after finishing my last game, which incidentally was winning a match, he dropped me a message, I also entered a message, then I hit "top page" and the game was still there. If I go back to the game and hit "next game" I get an "oops" crash (see mail).

Also I see a screen that only shows "active matches -1" (can send screenshot if needed)

**Reply by kagsteiner** [reply]
As the game was still in my list of open games, I went to the simulator to reproduce, but there the game vanished. :-(

**Reply by Smiles2go** [reply]
Sounds like what I am seeing. Have had as many as 4 games stuck in Top Page. Though I haven't seen active "-1".

Next time, visit DG website, login, look at those games. Do you see "(Predicted result)"?

I can clear those stuck games by going to DG web and hitting Next there. If more games in the match, they continue fine in DG App 1.4.4.

**Reply by hape42** [reply]
Good that someone else has the problem. This increases our chances of finding the cause.

I myself will now only play my matches in the iPhone simulator and hope that this will happen to me.

**Reply by hape42** [reply]
if this happens again, it would be most helpful if i could see the html code.

So: the error message comes. then leave everything as it is in the app and go to the computer. call up the match there in the browser.
From what appears there then I need the html code. 

in safari right click and follow the context menu.
mark the html code, copy it and send it to me by email to dg@hape42.de.

**Reply by Smiles2go** [reply]
I'm also getting the "Active Matches -1" but it only shows for a second or two before returning to Top Page. Here is screen shot:
![image](https://user-images.githubusercontent.com/121371957/211090275-f0b7dfd4-075b-499a-a818-cb503510f718.png)

**Reply by kagsteiner** [reply]
Yes this is exactly the faulty "active matches" I'm getting in this situation. Happened both on an iPad and later on an iPhone.

I've hopefully managed to send you the right HTML.

**Reply by kagsteiner** [reply]
Played games in the simulator today without any issue. Could even continue a game there that was stuck in my phone. Totally stunned:
- finish a game on my phone, get stuck as described above. Restarting the app on my phone, nothing helps, keeps stuck.
- switch to simulator and game continues nicely.???

**Reply by hape42** [reply]
![Bildschirm­foto 2023-01-07 um 09 26 25](https://user-images.githubusercontent.com/2091581/211141627-2939c71e-e2d0-4b6f-8eb0-3c64cf580b0c.png)

This is the trigger for the error message "unexpected....".

I have now solved this so that "next" is automatically typed.

I will finish a version today and distribute it via testFlight.

**Comment by hclhub** [comment]
I'll join the testing.  I use an iPad Air 3rd Generation running iOS 16  thortont@gmail.com

**Reply by hclhub** [reply]
1) Change in behavior for the Top Page and Refresh buttons (noticably) maybe others. They don't toggle colors as before when pressed.

When I have no matches to move, I would press the refresh (or top page).  I would know it was done with the refresh when the button toggled to reverse and then back to normal.  Now it doesn't toggle.  A refresh can sometimes take a while the first time, so I am not sure now if it has finished processing or not. Even if the performance on the first refresh wasn't so good before, subsequent presses were handled very quickly.  Have to wait til I get some matches where I can move queued up to check that vs previous.  Would like the color toggle on the buttons to be as they were before.

2) Toggling boards is very slow:when I click one it takes several seconds to move the check mark, and when I wasn't sure my press was registered, I pressed a few more times, and is was like it was frozen.  It cannot be scrolled.  Selecting an already selected board causes this temporary freeze as well.

**Comment by hape42** [comment]
done
